src/components/Editor/index.js

```diff
@@ -102,10 +102,21 @@ class Editor extends Component {
     if (this.props.onKeyDown) {
       this.props.onKeyDown(evt)
     }
+    
     if (evt.keyCode === 9 && !this.props.ignoreTabKey) { // Tab Key
-      document.execCommand('insertHTML', false, '&#009')
-      evt.preventDefault()
-    } else if (evt.keyCode === 13) { // Enter Key
+      if(!this.tabGuarded){
+        document.execCommand('insertHTML', false, '&#009')
+        evt.preventDefault()
+      }
+    }else if (evt.keyCode && evt.keyCode === 27) {//Esc Key
+      this.tabGuarded=true
```

Comment:
Also when you move the flag over to the state you might want to add a check for `!tabGuarded` here